### PR TITLE
Improve `backend/forum` tests

### LINKED ISSUE #3: Improve backend tests
Test coverage for the `backend` tests is about **57%**. We need to improve it to be at least **90%**.

### PR DESCRIPTION
**Pull request by maximihajlov**
This pull request finally resolves #3 by improving tests for the `backend/forum`. 

- Coverage was increased from 51% to 96%

- The tests now follow the black box methodology, there's no direct usage of DB functions. Only HTTP requests.

- The big `get_test` and `post_test` were split to several files testing the specific areas of the project. Each test file and each subtest is now independent, which makes it much easier to run only small set of tests while working on the specific feature (instead of running all the tests)

Before:
image

After:
image

This pull request also fixes some small bugs and imperfections found while testing. A few files were moved, but the behavior remains the same.

### CODE REVIEW

File: backend/forum/handlers/auth.go [hunk 1]
```diff
@@ -2,19 +2,47 @@ package handlers
...
+	MinLastNameLength = 1
+	MaxLastNameLength = MaxNameLength
+
+	MinPasswordLength = 8
+	MaxPasswordLength = 128
+
+	MinEmailLength = 3
+	MaxEmailLength = 254
+)
+
```
Comment (bomanviktor) [inline backend/forum/handlers/auth.go:36]: Good change. I prefer this way of declaring the constants.

File: backend/forum/handlers/test/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+			cli.TestPost(t, "/api/signup", badClientData, http.StatusBadRequest)
+
+			badClientData.Name = "THISUSERNAMEISTOOLONG"
+			cli.TestPost(t, "/api/signup", badClientData, http.StatusBadRequest)
+
+			badClientData.Name = "with spaces"
+			cli.TestPost(t, "/api/signup", badClientData, http.StatusBadRequest)
+
+		})
+
```
Comment (bomanviktor) [inline backend/forum/handlers/test/auth_test.go:53]: Very nice way to solve this. Very concise and easy to read and build upon.

File: backend/forum/handlers/test/users_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	cli1.TestAuth(t)
+
+	cli2 := testServer.TestClient()
+	cli2.TestAuth(t)
+
+	t.Run("Not found", func(t *testing.T) {
+		cli1.TestGet(t, "/api/user/10/posts", http.StatusNotFound)
+		cli1.TestGet(t, "/api/user/214748364712312214748364712312/posts", http.StatusNotFound)
+	})
+
```
Comment (bomanviktor) [inline backend/forum/handlers/test/users_test.go:67]: Why run this test twice? Shouldn't line 65 be enough?

**Comment by maximihajlov** [comment]
Deployed successfully. @bomanviktor, thanks for the review!

File: backend/forum/handlers/test/users_test.go (hunk 3 above)
Comment (maximihajlov) [reply backend/forum/handlers/test/users_test.go:67]: Nope. This case is to test `strconv.Atoi` error handling. This `userId` matches regexp `\d+`, but is not a valid integer (even int64), so atoi will fail for this case:

https://github.com/merpw/forum/blob/2abd855d960804ee2d0c3261babc28bf645fd7ea/backend/forum/handlers/user.go#L36-L46

Comment (bomanviktor) [reply backend/forum/handlers/test/users_test.go:67]: Oh! I see. Get outside of the range. I forgot what kind of safety measures were in the handler lol.

**Pull request merged**